packages/server/lib/projects.ts

```diff
@@ -0,0 +1,80 @@
+import browsers from './browsers'
+import { ProjectBase, Server } from './project-base'
+
+export type TestingType = 'e2e' | 'component'
+
+interface Args {
+  projectRoot: string
+  testingType: TestingType
+}
+
+class Projects {
+  private currentProject?: string
+  // key is projectRoot
+  projects: Record<string, ProjectBase<Server>> = {}
+
+  async addProject ({ testingType, projectRoot }: Args, { isCurrentProject }: { isCurrentProject: boolean }) {
```

Comment:
I could not get it to work, I probably am doing it wrong, but I found out I could this instead: `NexusGenInputs['AddProjectInput']`

Probably because the rest of the code is not using decorators yet, we should get it to use decorators.